### PR TITLE
wasmtime: 0.36.0 -> 0.37.0

### PR DESCRIPTION
**Pull request by ereslibre**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
@GrahamcOfBorg build wasmtime

**Review by ggreif — APPROVED** [review]

File: pkgs/development/interpreters/wasmtime/default.nix [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
 
-  cargoSha256 = "sha256-/+uioJRXiugsV7SUwsDNHGaPxrxrhscQUGyXOzzwG/g=";
+  cargoSha256 = "sha256-X+KDeWavFTBaxbSPlIiyuiBC7wg1/5C/NXp+VEY8Mk8=";
 
   # This environment variable is required so that when wasmtime tries
   # to run tests by using the rusty_v8 crate, it does not try to
   # download a static v8 build from the Internet, what would break
   # build hermetism.
-  RUSTY_V8_ARCHIVE = "${v8}/lib/libv8.a";
+  RUSTY_V8_ARCHIVE = lib.optionalString stdenv.isLinux "${v8}/lib/libv8.a";
```
Comment (ereslibre) [inline pkgs/development/interpreters/wasmtime/default.nix:21]: Force pushed to replace `lib.optionals` with `lib.optionalString`.

**Comment by happysalada** [comment]
The ci is happy, and I can't see anything to change.
Thank you for this!

**Comment by nomeata** [comment]
@ereslibre ,looks like wasmtime was removed on Darwin again, in: https://github.com/NixOS/nixpkgs/pull/173671. I wonder why…

**Comment by ereslibre** [comment]
@nomeata I will have a look at it; can see that hydra reports a failure on `x86_64-darwin` at https://hydra.nixos.org/build/187449906; apparently since wasmtime 0.37 back in May 22nd (https://hydra.nixos.org/build/177848118). I didn't realize this one and I actually removed the broken attribute in https://github.com/NixOS/nixpkgs/pull/178457 because I could not see any issue at all 🤦 

I will try to reproduce what hydra is reporting and mark the package as broken or try to fix the build...

**Comment by ereslibre** [comment]
This is probably an issue with the runtime trying to use certain CPU capabilities, errors in the log https://hydra.nixos.org/build/186536905/nixlog/1: `Unsupported feature: SIMD support requires SSE3, SSSE3, SSE4.1, and SSE4.2 on x86_64.'` in several places.

I will check later today and remove tests for x86_64-darwin with a comment on why tests are disabled. It's not an optimal solution because we lose that feedback on that platform, but the failure is very tied to the execution environment on x86_64-darwin on the hydra infrastructure.

**Pull request merged**